emsdk.py

```diff
@@ -1129,6 +1130,64 @@ def build_llvm_tool(tool):
   return success
 
 
+# LLVM git source tree migrated to a single repository instead of multiple ones, build_llvm_monorepo() builds via that repository structure
+def build_llvm_monorepo(tool):
+  debug_print('build_llvm_monorepo(' + str(tool) + ')')
+  fastcomp_root = tool.installation_path()
+  fastcomp_src_root = os.path.join(fastcomp_root, 'src')
+  success = git_clone_checkout_and_pull(tool.download_url(), fastcomp_src_root, tool.git_branch)
+  if not success:
+    return False
+
+  build_dir = fastcomp_build_dir(tool)
+  build_root = os.path.join(fastcomp_root, build_dir)
+
+  build_type = decide_cmake_build_type(tool)
+
+  # Configure
+  global BUILD_FOR_TESTING, ENABLE_LLVM_ASSERTIONS
+  tests_arg = 'ON' if BUILD_FOR_TESTING else 'OFF'
+
+  enable_assertions = ENABLE_LLVM_ASSERTIONS.lower() == 'on' or (ENABLE_LLVM_ASSERTIONS == 'auto' and build_type.lower() != 'release' and build_type.lower() != 'minsizerel')
+
+  if ARCH == 'x86' or ARCH == 'x86_64':
+    targets_to_build = 'WebAssembly;X86'
+  elif ARCH == 'arm':
+    targets_to_build = 'WebAssembly;ARM'
+  elif ARCH == 'aarch64':
+    targets_to_build = 'WebAssembly;AArch64'
+  else:
+    targets_to_build = 'WebAssembly'
+  args = ['-DLLVM_TARGETS_TO_BUILD=' + targets_to_build, '-DLLVM_INCLUDE_EXAMPLES=OFF', '-DCLANG_INCLUDE_EXAMPLES=OFF', '-DLLVM_INCLUDE_TESTS=' + tests_arg, '-DCLANG_INCLUDE_TESTS=' + tests_arg, '-DLLVM_ENABLE_ASSERTIONS=' + ('ON' if enable_assertions else 'OFF')]
+  # LLVM build system bug: looks like everything needs to be passed to LLVM_ENABLE_PROJECTS twice. (or every second field is ignored?)
+
+  # LLVM build system bug #2: compiler-rt does not build on Windows. It insists on performing a CMake install step that writes to C:\Program Files. Attempting
+  # to reroute that to build_root directory then fails on an error
+  #  file INSTALL cannot find
+  #  "C:/code/emsdk/llvm/git/build_master_vs2017_64/$(Configuration)/lib/clang/10.0.0/lib/windows/clang_rt.ubsan_standalone-x86_64.lib".
+  # (there instead of $(Configuration), one would need ${CMAKE_BUILD_TYPE} ?)
+  # It looks like compiler-rt is not compatible to build on Windows?
+  args += ['-DLLVM_ENABLE_PROJECTS="clang;clang;lld;lld"']
```

Comment:
I've never seen this before very strange.  Is this a windows-only problem?

Reply:
Not sure, haven't tested other platforms.